internal/auth/tokenloader_test.go

```diff
@@ -0,0 +1,309 @@
+
+	// T=0: First load - should read from file
+	token1, err := loader()
+	require.NoError(t, err)
+	assert.Equal(t, "my-secret-token", token1)
+
+	// Change the file content
+	updateTokenFile(t, tokenFile, "new-token\n")
+
+	// T=50ms: Still within cache interval (< 100ms)
+	currentTime = currentTime.Add(50 * time.Millisecond)
+	token2, err := loader()
+	require.NoError(t, err)
+	assert.Equal(t, "my-secret-token", token2, "Should return cached token within interval")
+
+	// T=150ms: Beyond cache interval (> 100ms)
+	currentTime = currentTime.Add(100 * time.Millisecond) // Total: 150ms
+	token3, err := loader()
+	require.NoError(t, err)
+	assert.Equal(t, "new-token", token3, "Should return refreshed token after cache expires")
+}
+
+// TestCachedFileTokenLoader_ExactBoundaries tests exact cache boundary conditions
+func TestCachedFileTokenLoader_ExactBoundaries(t *testing.T) {
+	currentTime := time.Unix(0, 0)
+	timeFn := func() time.Time { return currentTime }
+	tokenFile := createTempTokenFile(t, "boundary-token\n")
+	cacheInterval := 200 * time.Millisecond
+
+	loader := cachedFileTokenLoader(tokenFile, cacheInterval, timeFn)
+
+	// Load initial token
+	token, err := loader()
+	require.NoError(t, err)
+	assert.Equal(t, "boundary-token", token)
+
+	// Update file
+	updateTokenFile(t, tokenFile, "boundary-updated\n")
+
+	// Test exact boundary conditions
+	testCases := []struct {
+		timeAdvance   time.Duration
+		expectedToken string
+		description   string
+	}{
+		{199 * time.Millisecond, "boundary-token", "1ms before cache expires"},
+		{1 * time.Millisecond, "boundary-updated", "exactly at cache expiry"},
+		{50 * time.Millisecond, "boundary-updated", "well past cache expiry"},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			currentTime = currentTime.Add(tc.timeAdvance)
+			token, err := loader()
+			require.NoError(t, err)
+			assert.Equal(t, tc.expectedToken, token, tc.description)
+		})
+	}
+}
+
+// TestCachedFileTokenLoader_ZeroInterval tests disabled reloading
+func TestCachedFileTokenLoader_ZeroInterval(t *testing.T) {
+	currentTime := time.Unix(0, 0)
+	timeFn := func() time.Time { return currentTime }
+	tokenFile := createTempTokenFile(t, "initial-token\n")
+
+	loader := cachedFileTokenLoader(tokenFile, 0, timeFn) // Zero interval = no reloading
+
+	// T=0: Initial load
+	token, err := loader()
+	require.NoError(t, err)
+	assert.Equal(t, "initial-token", token)
+
+	// Update file content
+	updateTokenFile(t, tokenFile, "updated-token\n")
+
+	// Should still return cached token (no reloading)
+	currentTime = currentTime.Add(1 * time.Hour) // Advance time significantly
+	token, err = loader()
+	require.NoError(t, err)
+	assert.Equal(t, "initial-token", token, "Zero interval should disable reloading completely")
+
+	// Multiple calls should continue returning cached token
+	for i := 0; i < 5; i++ {
+		currentTime = currentTime.Add(1 * time.Hour)
```

Comment:
in addition to advancing time you may also overwrite the file every time with random value, both of these conditions will be ignored.